src/main/java/org/reaktivity/nukleus/http2/internal/Http2WriteScheduler.java

```diff
@@ -541,6 +541,11 @@ Http2Stream stream(int streamId)
     private void http2(Http2Stream stream, Http2FrameType type,
                        int sizeofGuess, Flyweight.Builder.Visitor visitor, boolean flush)
     {
+        // After RST_STREAM is written, don't write any frame in the stream
+        if (stream != null && type != RST_STREAM && stream.state == Http2Connection.State.CLOSED)
+        {
```

Comment:
Hoist this condition into a static method such as `canStreamWrite(stream, type)`.
Use condition to guard following block, rather than early `return`.